### PR TITLE
Remove MK_CHERI

### PR DESCRIPTION
**Pull request by brooksdavis**
MK_CHERI used to be the all-purpose way to say you're building with CHERI. We now have a more mature model using MACHINE_CPU and MACHINE_ABI and MK_CHERI is supposed to enable odd bits of CHERI-specific software. This PR removes the last uses that do otherwise.

Inspired by #1761

### CODE REVIEW

File: tests/sys/Makefile [hunk 1]
```diff
@@ -42,7 +42,7 @@ TESTS_SUBDIRS+=		vmm
 # during cleanup. They start and stop a new auditd for every test CASE which
 # takes a really long time on QEMU. In the latest jenkins run these tests took
 # 9.5 hours to run! Skip them until we can run them at a reasonable speed.
-.if ${MK_AUDIT} != "no" && ${MK_CHERI} == "no"
+.if ${MK_AUDIT} != "no" && ${MACHINE_CPU:Mcheri}
```
Comment (bsdjhb) [inline tests/sys/Makefile:45]: I think this needs to be `!${MACHINE_CPU:Mcheri}`?

**Review by bsdjhb — COMMENTED** [review]
Happy to lose the local diffs in sendmail makefiles.  Had a conflict when merging the upstream changes recently.

File: tests/sys/Makefile (hunk 1 above)
Comment (arichardson) [reply tests/sys/Makefile:45]: Hmm I feel like I fixed that upstream and we can drop this local diff.

Comment (arichardson) [reply tests/sys/Makefile:45]: I think https://github.com/CTSRD-CHERI/cheribsd/commit/df093aa9463b2121d8307fb91c4ba7cf17f4ea64 and https://github.com/CTSRD-CHERI/cheribsd/commit/869cc06480b75b4caea0d049e0cf7f82bb5aeed1 may have improved that situation.

**Comment by jrtc27** [comment]
Does MK_CHERI even serve a real purpose at this point? It's just gating libsyscalls and various things in bin. The only thing I could imagine is if libsyscalls is broken for non-CHERI architectures, but libsyscalls itself has nothing really to do with CHERI beyond being a thing that libcheri wanted to use.

**Comment by brooksdavis** [comment]
> Does MK_CHERI even serve a real purpose at this point? It's just gating libsyscalls and various things in bin. The only thing I could imagine is if libsyscalls is broken for non-CHERI architectures, but libsyscalls itself has nothing really to do with CHERI beyond being a thing that libcheri wanted to use.

Prodded by this comment I've removed the WITH(OUT)_CHERI option entirely.

**Review by arichardson — APPROVED** [review]

**Comment by brooksdavis** [comment]
This will conflict with #1761 and completing the change (with src.conf.5 updates) depends on that PR.

File: lib/Makefile [hunk 2]
```diff
@@ -171,7 +171,11 @@ SUBDIR.${MK_BLACKLIST}+=libblacklist
 SUBDIR.${MK_BLUETOOTH}+=libbluetooth libsdp
 SUBDIR.${MK_BSNMP}+=	libbsnmp
 
-SUBDIR.${MK_CHERI}+=	libsyscalls
+.if ${MACHINE_ABI:Mpurecap} || ${MK_LIB64C} == "yes"
```
Comment (jrtc27) [inline lib/Makefile:174]: MK_LIB64C doesn't feel right for lib, do you not want MACHINE_CPU:Mcheri?

**Review by bsdjhb — APPROVED** [review]
I'll rebase #1761 after merging this.

**Pull request merged**